### PR TITLE
added test for  urlhaus

### PR DESCRIPTION
**Pull request by Sekhar-Kumar-Dash**
## Changes proposed

Added more tests for  urlhaus

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

**Comment by AlyaGomaa** [comment]
Urlhaus Unit Tests should be added to .github/workflows/CI-production-testing.yml and .github/workflows/CI-staging.yml

**Comment by Sekhar-Kumar-Dash** [comment]
> Urlhaus Unit Tests should be added to .github/workflows/CI-production-testing.yml and .github/workflows/CI-staging.yml

I have already added it to production_testing.yml  just forgot about CI-staging.yml Sorry i will fix it

File: tests/test_urlhaus.py [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+def test_make_urlhaus_request(
+    mock_response, mock_db, to_lookup, expected_url, status_code
+):
+    """Test successful requests to the make_urlhaus_request function."""
+    mock_response_instance = Mock()
+    mock_response.return_value = mock_response_instance
+    mock_response = Mock(status_code=status_code)
+    mock_response_instance.post.return_value = mock_response
+
+    urlhaus = URLhaus(mock_db)
```
Comment (AlyaGomaa) [inline tests/test_urlhaus.py:42]: this should be moved to module_factory, always remember no object ever used in the tests should be created in a test_*.py file, because you'll need to recreate it again later and that would cause code duplication

Comment (AlyaGomaa) [reply tests/test_urlhaus.py:42]: check any object creation in all functions for the same issue ok?

File: tests/test_urlhaus.py [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+    )
+
+    urlhaus = URLhaus(mock_db)
+    urlhaus.urlhaus_session = mock_response_instance
+    with patch.object(
+        urlhaus, "create_urlhaus_session"
+    ) as mock_create_session:
+        mock_create_session.side_effect = requests.exceptions.ConnectionError
+        with pytest.raises(requests.exceptions.ConnectionError):
+            urlhaus.make_urlhaus_request(to_lookup)
```
Comment (AlyaGomaa) [inline tests/test_urlhaus.py:77]: what are you testing here? shouldn't we be checking that the return in None in case of a Connection error? also why did you add a side effect for create_urlhaus_session() ? isnt this `  mock_response_instance.post.side_effect ` enough?

File: tests/test_urlhaus.py [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+                "tags": "Trojan.Win32.Generic!BT",
+                "file_type": "PE32 executable (console) Intel 80386, for MS Windows",
+                "file_name": "malware.exe",
+            },
+            "a1b2c3d4",
+        ),
+    ],
+)
+@patch.object(URLhaus, "make_urlhaus_request")
+def test_parse_urlhaus_responses(
```
Comment (AlyaGomaa) [inline tests/test_urlhaus.py:169]: anything that comes after "test_" in the function name should be a function that exists in the class you're testing , right now there is no function called parse_urlhaus_responses()

File: tests/test_urlhaus.py (hunk 2 above)
Comment (Sekhar-Kumar-Dash) [reply tests/test_urlhaus.py:77]: @AlyaGomaa  I am so sorry for this is incomplete i was trying something else here.

File: tests/test_urlhaus.py (hunk 1 above)
Comment (Sekhar-Kumar-Dash) [reply tests/test_urlhaus.py:42]: Sorry i thought you were only asking about only the object which were being used as fixture in the code 
i am fixing it.

**Comment by Sekhar-Kumar-Dash** [comment]
hey @AlyaGomaa  sorry for the multiple commits please only review the last three PRs
